substrate/thread_pool

```diff
@@ -76,14 +76,14 @@ namespace substrate
 			}
 		}
 
-		result_t clearResultQueue() noexcept
+		inline result_t clearResultQueue() noexcept
 		{
 			result_t result{};
 			while (!results.empty())
 			{
 				auto thisResult = results.pop();
 				if (!result)
-					std::swap(result, thisResult);
+					result = thisResult;
```

Comment:
Does it make sense to mark this `std::move()` to invoke move assignment where possible?